draft-irtf-cfrg-vdaf.md

```diff
@@ -1424,89 +1405,55 @@ methods:
       Client.
     * `nonce` MUST have length `vdaf.NONCE_SIZE`.
 
+* `vdaf.prep_shares_to_prep(ctx: bytes, agg_param: AggParam, prep_shares:
+  list[PrepShare]) -> PrepMessage` is the deterministic preparation-message
+  pre-processing algorithm. It combines the prep shares produced by the
+  Aggregators in the previous round into the prep message consumed by each
+  Aggregator to start the next round.
+
 * `vdaf.prep_next(ctx: bytes, prep_state: PrepState, prep_msg: PrepMessage) ->
   tuple[PrepState, PrepShare] | OutShare` is the deterministic
   preparation-state update algorithm run by each Aggregator. It updates the
-  Aggregator's preparation state (`prep_state`) and returns either its next
-  preparation state and its message share for the current round or, if this is
-  the last round, its output share. An exception is raised if a valid output
-  share could not be recovered. The input of this algorithm is the inbound
-  preparation message.
+  Aggregator's prep state (`prep_state`) and returns either its next prep state
+  and prep share for the next round or, if this is the last round, its output
+  share.
 
-* `vdaf.prep_shares_to_prep(ctx: bytes, agg_param: AggParam, prep_shares:
-  list[PrepShare]) -> PrepMessage` is the deterministic preparation-message
-  pre-processing algorithm. It combines the prep shares generated by the
-  Aggregators in the previous round into the prep message consumed by each in
-  the next round.
-
-In effect, each Aggregator moves through a linear state machine with `ROUNDS`
-states.  The Aggregator enters the first state on using the initialization
-algorithm, and the update algorithm advances the Aggregator to the next state.
-Thus, in addition to defining the number of rounds (`ROUNDS`), a VDAF instance
-defines the state of the Aggregator after each round.
-
-The preparation-state update accomplishes two tasks: recovery of output shares
-from the input shares and ensuring that the recovered output shares are valid.
-The abstraction boundary is drawn so that an Aggregator only recovers an output
-share if it is deemed valid (at least, based on the Aggregator's view of the
-protocol). Another way to draw this boundary would be to have the Aggregators
-recover output shares first, then verify that they are valid. However, this
-would allow the possibility of misusing the API by, say, aggregating an invalid
-output share. Moreover, in protocols like Prio+ {{AGJOP21}} based on oblivious
-transfer, it is necessary for the Aggregators to interact in order to recover
-aggregatable output shares at all.
+An exception may be raised by one of these algorithms, in which case the report
+MUST be deemed invalid and not processed any further.
+
+Implementation note: The preparation-state update accomplishes two tasks:
```

Comment:
This phrasing doesn't make sense any more without the context of the preceding "linear state machine" paragraph
```suggestion
Implementation note: The preparation algorithm accomplishes two tasks:
```

Reply:
Changed to "process" since it's actually a collection of agorithms.